### PR TITLE
[Java][Spring] default value for list parameter

### PR DESCRIPTION
**Pull request by borsch**
Fix for https://github.com/OpenAPITools/openapi-generator/issues/11957

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
cc @cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02)

File: modules/openapi-generator/src/test/resources/bugs/issue_11957.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        schema:
+          type: integer
+          default: 0
+          minimum: 0
+      - in: query
+        name: orderBy
+        description: Filter by status not equal (AND condition applied)
+        schema:
+          type: array
+          default: ["updatedAt:DESC", "createdAt:DESC"]
```
Comment (welshm) [inline modules/openapi-generator/src/test/resources/bugs/issue_11957.yaml:29]: Could you add a test for an empty default as well?

```
default: []
```

**Review by welshm — COMMENTED** [review]
This looks great - I think I would just add a test for an empty default value if you're comfortable with that

File: modules/openapi-generator/src/test/resources/bugs/issue_11957.yaml (hunk 1 above)
Comment (borsch) [reply modules/openapi-generator/src/test/resources/bugs/issue_11957.yaml:26]: done

**Review by welshm — APPROVED** [review]

**Review by atextor — APPROVED** [review]
Looks good to me!
